src/StmtToViz.cpp

```diff
@@ -2851,123 +3043,162 @@ class IRVisualizer {
-                           </button>
-                         </div>
-                         <div>
-                           <button class='icon-btn resize-btn' onclick='collapseR_visualization_tab()'>
-                             <i class='bi bi-arrow-bar-right' title='Collapse visualization tab'></i>
-                           </button>
-                         </div>
-                       </div>
-                     </div>)";
-    }
-
-    // Generate a resizing bar to control the width of visualization and assembly tabs
-    void generate_resize_bar_2() {
-        stream << R"(<div class='resize-bar' id='resize-bar-2'>
-                       <div class='collapse-btns'>
-                         <div>
-                           <button class='icon-btn resize-btn' onclick='collapseL_visualization_tab()'>
-                             <i class='bi bi-arrow-bar-left' title='Collapse visualization tab'></i>
-                           </button>
-                         </div>
-                         <div>
-                           <button class='icon-btn resize-btn' onclick='collapse_assembly_tab()'>
-                             <i class='bi bi-arrow-bar-right' title='Collapse assembly tab'></i>
-                           </button>
-                         </div>
-                       </div>
-                     </div>)";
-    }
-
-    // Loads and initializes the javascript template from `ir_visualizer / javascript_template.html`
-    void generate_javascript() {
-        stream << halide_html_template_StmtToViz_javascript;
+    // Generate pane: Generated device code
+    void generate_device_code_pane(const Buffer<> &buf) {
+        stream << "<div id='device-code-pane' class='pane'>\n";
+        int length = buf.size_in_bytes();
+        while (length > 0 && ((const char *)buf.data())[length - 1] == '\0') {
+            length--;
+        }
+        std::string str((const char *)buf.data(), length);
+        if (starts_with(buf.name(), "cuda_")) {
+            html_code_printer.print_cuda_gpu_source_kernels(str);
+        } else {
+            stream << "<code>\n";
+            stream << str;
+            stream << "\n";
+            stream << "</code>\n";
+        }
+        stream << "</div>\n";
+    }
+
+    // Generate a resizing bar to control the width of code and visualization panes
+    void generate_resize_bar(int num) {
+        stream << "<div class='resize-bar' id='resize-bar-" << num << "'>\n";
+        stream << "   <div class='collapse-btns'>\n";
+        stream << "     <div>\n";
+        stream << "      <button class='collapse-left' onclick='collapseTab(" << num << ")' title='Collapse pane on the left'></span>\n";
+        stream << "       </button>\n";
+        stream << "     </div>\n";
+        stream << "     <div>\n";
+        stream << "       <button class='collapse-right' onclick='collapseTab(" << (num + 1) << ")' title='Collapse pane on the right'></span>\n";
```

Comment:
Yeah, I was trying to help distinguish this by shading the button. My main concern is screen estate. The green seperator is already quite thick, and wasting screen estate. For now, I decided to not waste more space just to be able to put a bigger arrow icon. I never really looked at the icons anyway. I just picked a few unicode arrows, slammed them on, and realized I was looking at the color of the button instead of the icon.

Reply:
> A minor UX issue: Once I clicked the top button to collapse the stmt tab, the bottom button clears out the assembly tab.

Wait, that's exactly what it's supposed to do right? The button toggles the collapsedness of the tab left or right of it. Top button controls the tab on the left. Bottom button controls the tab on the right. You just click them to toggle whether they are collapsed or not.

Reply:
Marked as resolved.